### ISSUE TITLE
[BUG] Segmentation fault running an obfuscated PyArmor script

### ISSUE THREAD

**Issue by null-routed** [dashingsoft/pyarmor#1030]
### The issue
Python version: 3.9.6
Platform: Mac Intel
Pyarmor version: 8.0.19

I am building a python app and I'm using [Textual](https://github.com/Textualize/textual) as an API to build a simple terminal interface, but my code keeps terminating due to a seg fault after I obfuscate it.

### How to reproduce
Here's a simple code I took from one of their examples
```python
import sys

from rich.syntax import Syntax
from rich.traceback import Traceback

from textual import events
from textual.app import App, ComposeResult
from textual.containers import Container, VerticalScroll
from textual.reactive import var
from textual.widgets import DirectoryTree, Footer, Header, Static

class CodeBrowser(App):

    BINDINGS = [
        ("f", "toggle_files", "Toggle Files"),
        ("q", "quit", "Quit"),
    ]

    show_tree = var(True)

    def watch_show_tree(self, show_tree: bool) -> None:
        """Called when show_tree is modified."""
        self.set_class(show_tree, "-show-tree")

    def compose(self) -> ComposeResult:
        """Compose our UI."""
        path = "./" if len(sys.argv) < 2 else sys.argv[1]
        yield Header()
        with Container():
            yield DirectoryTree(path, id="tree-view")
            with VerticalScroll(id="code-view"):
                yield Static(id="code", expand=True)
        yield Footer()

    def on_mount(self, event: events.Mount) -> None:
        self.query_one(DirectoryTree).focus()

    def on_directory_tree_file_selected(
        self, event: DirectoryTree.FileSelected
    ) -> None:
        event.stop()
        code_view = self.query_one("#code", Static)
        try:
            syntax = Syntax.from_path(
                event.path,
                line_numbers=True,
                word_wrap=False,
                indent_guides=True,
                theme="github-dark",
            )
        except Exception:
            code_view.update(Traceback(theme="github-dark", width=None))
            self.sub_title = "ERROR"
        else:
            code_view.update(syntax)
            self.query_one("#code-view").scroll_home(animate=False)
            self.sub_title = event.path

    def action_toggle_files(self) -> None:
        self.show_tree = not self.show_tree

if __name__ == "__main__":
    CodeBrowser().run()
```
    

`python3 myExample.py` -> Code runs as expected
`pyarmor gen myExample.py` -> Code gets obfuscated correctly, no errors
`python3 dist/myExample.py` -> `zsh: segmentation fault  python3 dist/myExample.py`

**Comment by jondy** [comment]
Got it, but it will be checked later because it uses third party package `Textual`.

It's better to reproduce this issue with official Python library.

**Comment by jondy** [comment]
Try it in 8.0.20, it fixed a bug regarding class def.

**Comment by null-routed** [comment]
> Got it, but it will be checked later because it uses third party package `Textual`.
> 
> It's better to reproduce this issue with official Python library.

Understood. In the meanwhile I will try to track down the issue and reproduce it without Textual.

> Try it in 8.0.20, it fixed a bug regarding class def.

Issue still present in 8.0.20. Thanks for the update though.

**Comment by null-routed** [comment]
Actually I was wrong, I did not notice that my upgrade to 8.0.20 had failed due to an error. I fixed the error and successfully upgraded to 8.0.20 and I can confirm 8.0.20 did fix the issue. Thank you @jondy .
